frontend/src/scenes/actions/actionEditLogic.js

```diff
@@ -0,0 +1,85 @@
+import { kea } from 'kea'
+import api from 'lib/api'
+import { uuid } from 'lib/utils'
+import { toast } from 'react-toastify'
+
+export const actionEditLogic = kea({
+    key: props => props.id || 'new',
+    actions: () => ({
+        saveAction: true,
+        setAction: action => ({ action }),
+        setCreateNew: createNew => ({ createNew }),
+        setErrorActionId: actionId => ({ actionId }),
+    }),
+
+    loaders: ({ props }) => ({
+        action: {
+            loadAction: async () => {
+                return await api.get(props.apiURL + 'api/action/' + props.id)
+            },
+        },
+    }),
+
+    reducers: () => ({
+        action: [
+            null,
+            {
+                setAction: (_, { action }) => action,
+            },
+        ],
+        errorActionId: [
+            null,
+            {
+                setErrorActionId: (_, { actionId }) => actionId,
+            },
+        ],
+        createNew: [
+            false,
+            {
+                setCreateNew: (_, { createNew }) => createNew,
+            },
+        ],
+    }),
+
+    listeners: ({ values, props, actions }) => ({
+        saveAction: async () => {
+            let action = { ...values.action }
+            actions.setErrorActionId(null)
```

Comment:
I think it's cleaner if instead of using `setErrorActionId(null)` and later `actions.setErrorActionId(response.id)`, we'd set the `errorActionId` reducer to null with the `saveAction` action directly... and later have a more contextually meaningful action like `actions.actionAlreadyExists(response.id)`, which would set the `errorActionId` to `response.id`. This makes it much easier to see how the values change their state.

I had a similar thought when looking at `actionLogic` above. `isComplete` could be set to `false` when `loadAction` is called in the reducer, not explicitly in the listener.

The [kea docs](https://kea.js.org/docs/guide/concepts#listeners) write about this (search for "If you read the reducers section above, you'll remember that it's an anti-pattern to only have setThis and setThat actions that only update this or that.")